### PR TITLE
[teleop] Add camera selection boxes to Auton GUI

### PR DESCRIPTION
**Pull request by bli76**
image
image

Also updated Camera.vue and CameraSelection.vue from the ik-integration branch

### CODE REVIEW

File: base_station/gui/src/components/AutonTask.vue [hunk 1]
```diff
@@ -44,14 +44,16 @@
     <div class="box waypoints light-bg">
       <AutonWaypointEditor v-bind:odom="odom" v-bind:AutonDriveControl="AutonDriveControl"/>
     </div>
-    <div class="box angles light-bg">
+    <div class="box cameras light-bg">
+      <AutonCameras v-if="!this.autonEnabled" />
```
Comment (CameronTressler) [inline base_station/gui/src/components/AutonTask.vue:48]: Did you test this functionality with the big auton enabled button in the top right of the GUI?

File: base_station/gui/src/conf.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
 {
 	"cameras":
 	{
-		"0": "Microscope",
+        "-1": "No Camera",
+		"0": "Camera 0",
```
Comment (CameronTressler) [inline base_station/gui/src/conf.json:5]: Can you fix the formatting and add a new line to this file?

File: rover_msgs/Cameras.lcm [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package rover_msgs;
+
+struct Cameras {
+    int8_t port_0;
+    int8_t port_1;
+}
```
Comment (CameronTressler) [inline rover_msgs/Cameras.lcm:6]: Can you add a new line to this file

File: base_station/gui/src/components/AutonTask.vue (hunk 1 above)
Comment (bli76) [reply base_station/gui/src/components/AutonTask.vue:48]: yeah, works as intended

Comment (CameronTressler) [reply base_station/gui/src/components/AutonTask.vue:48]: coolio

**Review by CameronTressler — APPROVED** [review]
Updated marker looking clean

**Comment by cgiger00** [comment]
does this show both cameras? will we ever try to stream more than one?

**Comment by CameronTressler** [comment]
> does this show both cameras? will we ever try to stream more than one in auton?

Yeah this is my bad Bryan, I misremembered what we wanted for auton. Can you remove one of the CameraSelection modules? Then for port_1, just always send -1

File: base_station/gui/src/conf.json (hunk 2 above)
Comment (cgiger00) [reply base_station/gui/src/conf.json:5]: was this resolved?

**Review by cgiger00 — APPROVED** [review]

**Comment by CameronTressler** [comment]
@cgiger00 do you think we should have two cameras/is it urgent to merge in? Otherwise we can wait till just one camera selection is present

**Comment by bli76** [comment]
> @cgiger00 do you think we should have two cameras/is it urgent to merge in? Otherwise we can wait till just one camera selection is present

Yeah it's a simple change that I can make probably sometime tmrw

**Comment by cgiger00** [comment]
one camera sounds fine to me, i think the most we'll need is either the zed or the mast cam but probably not at the same time

File: base_station/gui/src/components/AutonCameras.vue [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                this.cam_index = CAMERA_NUM['up_middle_button']
+              } else if (gamepad.buttons[JOYSTICK_CONFIG['down_right_button']]['pressed']) {
+                this.cam_index = CAMERA_NUM['down_right_button']
+              } else if (gamepad.buttons[JOYSTICK_CONFIG['up_right_button']]['pressed']) {
+                this.cam_index = CAMERA_NUM['up_right_button']
+              }
+            }
+          }
+        }
+        this.$parent.publish('/microscope', {type: "Microscope", streaming: this.microscope_cam === true})
```
Comment (CameronTressler) [inline base_station/gui/src/components/AutonCameras.vue:88]: Don't need to publish to microscope

File: base_station/gui/src/components/AutonCameras.vue [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      }, 250)
+    },
+
+    methods: {
+      setCamIndex: function (new_index) {
+        this.cam_index = new_index
+      },
+
+      toggleMicroscopeCam: function () {
```
Comment (CameronTressler) [inline base_station/gui/src/components/AutonCameras.vue:97]: Similarly, can be removed

**Comment by bli76** [comment]
also updated formatting

image

**Pull request merged**